src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -1493,13 +1493,20 @@ private GHPullRequest getDetailedGHPullRequest(int number, TaskListener listener
 
     private void ensureDetailedGHPullRequest(GHPullRequest pr, TaskListener listener, GitHub github, GHRepository ghRepository) throws IOException, InterruptedException {
         final long sleep = 1000;
-        while (pr.getMergeableState() == null) {
+        int retryCountdown = 4;
+
+        Connector.checkApiRateLimit(listener, github);
+        while (pr.getMergeable() == null && retryCountdown > 1) {
```

Comment:
mergeableState can now return "unknown" instead of null. :sob: 
Switched checking mergeable which is null until it is true or false.